File: packages/kevin_flutter_accounts/kevin_flutter_accounts_ios/ios/Classes/Model/KevinBank.swift
import Foundation
import kevin_ios

internal struct KevinBank: Codable {
    let id: String
    let name: String
    let officialName: String?
    let imageUri: String
    let bic: String?
}

internal extension ApiBank {
    func toKevinBank() -> KevinBank {
        return KevinBank(
            id: self.id,
            name: self.name,
            officialName: self.officialName,
            imageUri: self.imageUri,
            bic: self.bic)
    }
}

File: packages/kevin_flutter_accounts/kevin_flutter_accounts_platform_interface/lib/src/model/kevin_bank.dart
class KevinBank {
  final String id;
  final String name;
  final String? officialName;
  final String imageUri;
  final String? bic;

  const KevinBank({
    required this.id,
    required this.name,
    required this.officialName,
    required this.imageUri,
    required this.bic,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'officialName': officialName,
      'imageUri': imageUri,
      'bic': bic,
    };
  }

  factory KevinBank.fromMap(Map<String, dynamic> map) {
    return KevinBank(
      id: map['id'] as String,
      name: map['name'] as String,
      officialName: map['officialName'] as String?,
      imageUri: map['imageUri'] as String,
      bic: map['bic'] as String?,
    );
  }
}
